### ISSUE TITLE
GeoImage.open doesn't seem to follow docs

### ISSUE THREAD

**Issue by ra-tolson** [gipit/gippy#171]
Looks like either the API has drifted or else the docs need updating.  The docs say you can go:

```
geoimg = GeoImage.open(filename, bandnames=(['red', 'green', 'blue']), nodata=0)
```

But when I try it it looks like it's iterating over the filename's characters:

```
(Pdb) gippy.GeoImage.open('foo.tif')
*** RuntimeError: 4: f: No such file or directory
```

I can workaround with a list though:

```
(Pdb) gippy.GeoImage.open(['foo.tif'])
*** RuntimeError: 4: foo.tif: No such file or directory
```

The reason I'd _like_ to call `GeoImage.open()` over `GeoImage()` is that with the former I can set nodata & gain as part of the call; I don't see how to do that with `GeoImage()`.

**Comment by matthewhanson** [comment]
Yes, it looks like the docs are wrong. `GeoImage.open` accepts a list of filenames. If you provide a string it iterates over it like it was a list, which returns the individual characters. 

It's true you can't set those parameters with the normal constructor. It's because the regular constructor essentially opens up the dataset as GDAL does it, whereas the nodata and gain are set after the dataset is open, so that is in the higher-level convenience function `GeoImage.open`.
